src/utils.ts

```diff
@@ -96,19 +96,23 @@ export function moveFootnotesToEnd(text: string) {
 }
 
 /**
- * Substitutes YAML and codeblocks in a text with a placeholder.
+ * Substitutes YAML, links, and codeblocks in a text with a placeholder.
  * Then applies the given function to the text.
  * Substitutes the YAML and codeblocks back to their original form.
  * @param {string} text - The text to process
  * @param {function(string): string} func - The function to apply to the text
  * @return {string} The processed text
  */
-export function ignoreCodeBlocksAndYAML(text: string, func: (text: string) => string): string {
+export function ignoreCodeBlocksYAMLAndLinks(text: string, func: (text: string) => string): string {
   const codePlaceholder = 'PLACEHOLDER 321417';
   const ret = replaceCodeblocks(text, codePlaceholder);
   text = ret.text;
   const replacedCodeBlocks = ret.replacedCodeBlocks;
 
+  const linkPlaceHolder = 'PLACEHOLDER_LINK 57849';
```

Comment:
I think we should replace YAML before links, if there is a link in the YAML we dont want to touch it at all.

Reply:
Sounds good. I have swapped the order.